### PR TITLE
[ricos-plugins] external alignment button

### PR DESCRIPTION
**Pull request by noamtwix**

### CODE REVIEW

File: packages/ricos-plugins/web/src/plugin-text-button.ts [hunk 1]
```diff
@@ -168,7 +176,7 @@ export class PluginTextButtons implements FormattingToolbarButtons {
     //TODO: support all buttons
     const unsupportedTextButtons = [
       'fontSize',
-      'alignment',
+      // 'alignment',
```
Comment (noamtwix) [inline packages/ricos-plugins/web/src/plugin-text-button.ts:179]: delete

File: packages/toolbars-modals/web/libs/getSelectionTextAlignment/package.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (noamtwix) [inline packages/toolbars-modals/web/libs/getSelectionTextAlignment/package.json:1]: delete file

File: packages/toolbars-modals/web/src/alignment/AlignmentPanelController.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
```
Comment (noamtwix) [inline packages/toolbars-modals/web/src/alignment/AlignmentPanelController.tsx:1]: types

File: packages/toolbars-modals/web/src/utils.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+};
+
+export const hasStyleChanges = (
+  currentHeading: string,
+  inlineStyles: Record<string, string>,
+  documentStyle?: DocumentStyle
+) => {
+  const headerStyle = documentStyle?.[DRAFT_TO_DOC_TYPE_WITH_LISTS[currentHeading]] || {};
+  return Object.entries(inlineStyles).some(([key, value]) => headerStyle[key] !== value);
+};
```
Comment (noamtwix) [inline packages/toolbars-modals/web/src/utils.tsx:16]: delete

File: packages/toolbars-modals/web/src/alignment/consts.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  {
+    text: 'AlignTextJustifyButton_Tooltip',
+    tooltip: 'AlignTextJustifyButton_Tooltip',
+    tooltipShortcut: {
+      MacOS: ' (⌘⇧J)',
+      Windows: ' (Ctrl+⇧+J)',
+    },
+    commandKey: 'justify',
+    icon: AlignJustifyIcon,
+    dataHook: 'textAlignmentButton_justify',
```
Comment (maimonav) [inline packages/toolbars-modals/web/src/alignment/consts.tsx:46]: @lxgreen I remember you talked about a new API for shortcuts, right?

File: packages/toolbars-modals/web/src/alignment/utils.tsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import { AlignLeftIcon, AlignTextCenterIcon, AlignRightIcon, AlignJustifyIcon } from './icons';
+import type { EditorCommands, TextAlignment } from 'ricos-types';
+
+export const getDefaultAlignment = (editorCommands: EditorCommands | void): TextAlignment => {
+  const selectedAlignment = editorCommands?.getTextAlignment();
+  return selectedAlignment || 'left';
+};
```
Comment (maimonav) [inline packages/toolbars-modals/web/src/alignment/utils.tsx:7]: Looks redundant, you can simply return 'left' in line 10, or am I missing something?

File: packages/toolbars-modals/web/src/panels/DesktopPanel.tsx [hunk 7]
```diff
@@ -22,6 +23,17 @@ const DesktopPanel = ({
   t,
   displayIconAndText,
   externalFocus,
+}: {
+  currentSelect?: any;
+  options?: any;
+  onChange?: any;
+  customPanelOptions?: any;
+  customPanelName?: string;
+  theme?: any;
```
Comment (maimonav) [inline packages/toolbars-modals/web/src/panels/DesktopPanel.tsx:32]: theme: any => RicosTheme?

File: packages/toolbars-modals/web/src/panels/DesktopPanel.tsx [hunk 8]
```diff
@@ -22,6 +23,17 @@ const DesktopPanel = ({
   t,
   displayIconAndText,
   externalFocus,
+}: {
+  currentSelect?: any;
```
Comment (maimonav) [inline packages/toolbars-modals/web/src/panels/DesktopPanel.tsx:27]: is there a type for other `any` type fields?

**Review by maimonav — CHANGES_REQUESTED** [review]

File: packages/ricos-types/web/src/modalTypes.ts [hunk 9]
```diff
@@ -27,6 +27,7 @@ export interface ModalConfig {
   Component: ComponentType;
   id: string;
   shortcuts?: KeyboardShortcut[];
+  layout?: Layout;
 }
```
[the remark this answers is not in the document]
Comment (noamtwix) [reply packages/ricos-types/web/src/modalTypes.ts:31]: yes

File: packages/ricos-plugins/web/src/plugin-text-button.ts [hunk 10]
```diff
@@ -104,9 +104,10 @@ export class PluginTextButton implements FormattingToolbarButton {
       type: 'button',
       tooltip: this.getTooltip(),
       toolbars: [],
-      getIcon: () => this.button.presentation?.icon,
+      getIcon: () =>
+        this.button.presentation?.icon || this.button.presentation?.getIcon(editorCommands),
       getLabel: () => this.button.id,
-      onClick: () => {
+      onClick: e => {
```
[the remark this answers is not in the document]
Comment (noamtwix) [reply packages/ricos-plugins/web/src/plugin-text-button.ts:110]: done

File: packages/toolbars-modals/web/src/alignment/utils.tsx (hunk 6 above)
Comment (noamtwix) [reply packages/toolbars-modals/web/src/alignment/utils.tsx:7]: done

File: packages/ricos-plugins/web/src/plugin-text-button.ts [hunk 11]
```diff
@@ -120,7 +120,7 @@ export class PluginTextButton implements FormattingToolbarButton {
                   this.button.modal.layout === 'toolbar'
                     ? {
                         placement: 'bottom',
-                        referenceElement: e.target,
+                        referenceElement: e?.target,
```
Comment (maimonav) [inline packages/ricos-plugins/web/src/plugin-text-button.ts:123]: If e is not passed, but the modal.layout === 'toolbar', what is the expected behavior?

Comment (noamtwix) [reply packages/ricos-plugins/web/src/plugin-text-button.ts:123]: the model will be in the top left corner of the window

**Pull request merged**